### DISCUSSION TITLE
FTP motion

### DISCUSSION THREAD

**Discussion by Kabre555** [Sunoo/homebridge-camera-ffmpeg#1121] [General]
I have a video doorbell camera that supports FTP upload on alert, but it doesn't allow me to select a filename without having the time put into the name.  I'm also not sure what to put in the path , tried root folder but keep getting an error "failed to create directory"

what is the correct way to install this, and is there a way to maybe change the trigger to be the username of the ftp user instead of a filename?

**Comment by Sunoo** [comment]
The filename doesn’t matter, it’s done by the path. The path should be the name of your camera.

**Reply by Kabre555** [reply]
not sure how, but I got it working..   the docker image, and vm machine image distro's have firewalls and the root pw was not supplied on the VM, anyway has to get it going on Raspberry ..  also didn't work with random login to ftp, had to use anonymous settings

**Comment by Kabre555** [comment]
Hi, still having problems with permissions.. can you tell me what is the full path for the FTP folder that the plugin uses

I can ftp in, and see camera folders but I can not switch into them.  

thx

**Reply by Sunoo** [reply]
The full path is literally just the camera name. It’s a virtual file system, so there is no more than that. Not sure why you can’t open the folders, but they’d be empty anyways.

**Reply by Kabre555** [reply]
ok, I'm about to start all over again.. so far I can only get this to work with a pi3 .. docker images dont work with your ftp plugin.. the smtp works fine.  but I need FTP.. can you recommend a solid platform or scratch install with a flavor of linux and I'll build from scratch.   also do you know if your FTP plugin works with the HOOBS platform ?

thx

**Reply by Sunoo** [reply]
I personally do my testing on Raspberry Pi OS, but it should work on any OS. I’m guessing it can work in Docker too, I just have no idea what settings would be needed to open the port.

It should work fine under HOOBS, but I have not tested current versions on it myself. I do plan to test all future versions under both Homebridge and HOOBS though.

**Reply by Kabre555** [reply]
do you know if your plugin's work on home assistant?

**Reply by Sunoo** [reply]
I’m not very familiar with Home Assistant, but I was under the impression their plugin system was completely incompatible with Homebridge.
